src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -440,23 +473,51 @@ void activity_classifier::perform_training_iteration() {
     float batch_loss = std::accumulate(loss_batch.data(),
                                        loss_batch.data() + loss_batch.size(),
                                        0.f, std::plus<float>());
+
+    
     cumulative_batch_loss += batch_loss / batch.batch_info.size();
 
+    
+    const shared_float_array& output = results.at("output");
+    float cumulative_per_batch_accuracy = 0.f;
+    
+    for (size_t i = 0; i < batch.batch_info.size(); ++i) {
```

Comment:
Could you create a second helper function that computes the accuracy for a given training batch? This second helper function can call the first one, and help modularize the code further